### ISSUE TITLE
Fix S4143: False Positive when variable is reassigned

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#1643]
### Description

S4143 Rule causes a lot of issues throughout our application analyses, however, most of them are false positive. This rule is unable to detect object c

### Repro steps

```csharp
dr["CurrencyCode"] = General.GeneralParametersFinder.LocalCurrencyCode();
dt.Rows.Add(dr);
dr = dt.NewRow();
dr["CurrencyCode"] = Framework.GeneralParameters.E_U_R;  // False issue raised here. dr refers to a new DataRow because of the previous line
```

### Related information

From https://community.sonarsource.com/t/sonarc-7-3-collection-elements-should-not-be-replaced-unconditionally-false-positive-problem/1622
